src/db/models/gnosis_safe_withdrawal.rs

```diff
@@ -321,4 +337,80 @@ mod tests {
             .is_none()
         );
     }
+
+    #[test_with_logger]
+    fn test_find_unaudited_withdrawals(logger: Logger) {
+        let mut rng = mc_util_test_helper::get_seeded_rng();
+        let test_db_context = TestDbContext::default();
+        let reserve_auditor_db = test_db_context.get_db_instance(logger.clone());
+        let token_id = TokenId::from(1);
+        let conn = reserve_auditor_db.get_conn().unwrap();
+
+        // Create two BurnTxOuts.
+        let burn_tx_out1 = create_and_insert_burn_tx_out(token_id, 100, &conn, &mut rng);
+        let burn_tx_out2 = create_and_insert_burn_tx_out(token_id, 100, &conn, &mut rng);
+
+        // Create two Gnosis withdrawals.
+        let mut withdrawal1 =
+            create_gnosis_safe_withdrawal_from_burn_tx_out(&burn_tx_out1, &mut rng);
+        let mut withdrawal2 =
+            create_gnosis_safe_withdrawal_from_burn_tx_out(&burn_tx_out2, &mut rng);
+
+        // Since they haven't been inserted yet, they should not be found.
+        assert!(GnosisSafeWithdrawal::find_unaudited_withdrawals(&conn)
+            .unwrap()
+            .is_empty());
+
+        // Insert the first withdrawal, it should now be found.
+        insert_gnosis_withdrawal(&mut withdrawal1, &conn);
+
+        assert_eq!(
+            GnosisSafeWithdrawal::find_unaudited_withdrawals(&conn).unwrap()[0],
+            withdrawal1
```

Comment:
Comparing the actual contents of the vec would be better, e.g. 
```
assert_eq!(
            GnosisSafeWithdrawal::find_unaudited_withdrawals(&conn).unwrap(),
            vec![withdrawal1.clone()],
);
```
Same for other places where you just check the length, can they be changed to comparing to a `vec![..]`?